templates/src/main/java/org/teavm/flavour/templates/parsing/Parser.java

```diff
@@ -171,7 +171,7 @@ private String parseText(int start, int end) {
                 break;
             }
             sb.append(source.subSequence(start, ref.getBegin()));
-            sb.append(ref.getChar());
+            sb.append(Character.toChars(ref.getCodePoint()));
```

Comment:
What about `sb.appendCodePoint(ref.getCodePoint())`?

Reply:
Even better!  I will try it out right now.

Reply:
OK, it works great, thanks!  I have checked in the fix.